volatility3/framework/symbols/linux/utilities/tainting.py

```diff
@@ -0,0 +1,133 @@
+from volatility3 import framework
+from volatility3.framework import interfaces
+from volatility3.framework.symbols.linux.utilities import LinuxUtilityInterface
+from volatility3.framework.constants import linux as linux_constants
+from typing import List, Optional
+
+
+class Tainting(LinuxUtilityInterface):
+    """Tainted kernel and modules parsing capabilities.
+
+    Relevant Linux kernel functions:
+        - modules: module_flags_taint
+        - kernel: print_tainted
+    """
+
+    _version = (1, 0, 0)
+    _required_framework_version = (2, 14, 0)
+
+    framework.require_interface_version(*_required_framework_version)
+
+    def __init__(
+        self,
+        context: interfaces.context.ContextInterface,
+        kernel_module_name: str,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self._kernel = context.modules[kernel_module_name]
+
+    @property
+    def _kernel_taint_flags_list(
+        self,
+    ) -> Optional[List[interfaces.objects.ObjectInterface]]:
+        if self._kernel.has_symbol("taint_flags"):
+            return list(self._kernel.object_from_symbol("taint_flags"))
+        return None
+
+    def _module_flags_taint_pre_4_10_rc1(
+        self, taints: int, is_module: bool = False
+    ) -> str:
+        """Convert the module's taints value to a 1-1 character mapping.
+        Relies on statically defined taints mappings in the framework.
+
+        Args:
+            taints: The taints value, represented by an integer
+            is_module: Indicates if the taints value is associated with a built-in/LKM module
+
+        Returns:
+            The raw taints string.
+        """
+        taints_string = ""
+        for char, taint_flag in linux_constants.TAINT_FLAGS.items():
+            if is_module and is_module != taint_flag.module:
+                continue
+
+            if taints & taint_flag.shift:
+                taints_string += char
+
+        return taints_string
+
+    def _module_flags_taint_post_4_10_rc1(
+        self, taints: int, is_module: bool = False
+    ) -> str:
+        """Convert the module's taints value to a 1-1 character mapping.
+        Relies on kernel symbol embedded taints definitions.
+
+            struct taint_flag {
+                char c_true;		/* character printed when tainted */
+                char c_false;		/* character printed when not tainted */
+                bool module;		/* also show as a per-module taint flag */
+            };
+
+        Args:
+            taints: The taints value, represented by an integer
+            is_module: Indicates if the taints value is associated with a built-in/LKM module
+
+        Returns:
+            The raw taints string.
+        """
+        taints_string = ""
+        for i, taint_flag in enumerate(self._kernel_taint_flags_list):
```

Comment:
Using a more descriptive name like `taint_bit` instead of `i` would make the code clearer.